core/trino-spi/src/main/java/io/trino/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -1740,6 +1739,8 @@ private static void copyUnscaledDecimal(Slice from, Slice to)
 
     private static void pack(Slice decimal, int v0, int v1, int v2, int v3, boolean negative)
     {
+        // check if the result is zero and if so, don't create negative zeros:
+        negative &= v0 != 0 | v1 != 0 | v2 != 0 | v3 != 0; // use | instead of || to avoid branches
```

Comment:
`// use | instead of || to avoid branches`

Does it really avoid branches or it's hypothesis? Why would JIT not optimize `||` on booleans?

Reply:
I wanted to verify that. I couldn't find a way for the JVM to print the assembly instead of raw binary code, though. Do you know one?

Reply:
> I wanted to verify that. I couldn't find a way for the JVM to print the assembly instead of raw binary code, though. Do you know one?

You have to use asm extension while running JMH on linux

Reply:
I came to a point where I was told to compile a native library and put it into JDK's directory somewhere, at which point I was too tired of all this and gave up ;) I'm pretty sure the performance impact of this line is negligible either way, though, as it's trivial compared to all the other code that's executed for each operation.

Reply:
Sure, changed.